src/contract/HandlerBasedContract.ts

```diff
@@ -989,4 +968,37 @@ export class HandlerBasedContract<State> implements Contract<State> {
 
     return result as HandlerBasedContract<unknown>;
   }
+
+  // Call contract and verify if there are any internal writes:
+  // 1. Evaluate current contract state
+  // 2. Apply input as "dry-run" transaction
+  // 3. Verify the callStack and search for any "internalWrites" transactions
+  // 4. For each found "internalWrite" transaction - generate additional tag:
+  // {name: 'InternalWrite', value: callingContractTxId}
+  private async handleInternalWrites<Input>(
```

Comment:
`evaluateInteranalWrites`? (is a bit misleading..)
`discoverInternalWrites` (sounds a bit silly...)
`generateInternalWrites` 

..?